Framework/API/src/AlgorithmFactory.cpp

```diff
@@ -234,13 +235,10 @@ int AlgorithmFactoryImpl::highestVersion(const std::string &algorithmName) const
   else {
     // Fall back, algorithmName might be an alias
     // Check alias map, then find version from real name
-    const auto realName = getRealNameFromAlias(algorithmName);
-    if (realName != boost::none) {
-      viter = m_vmap.find(realName.get());
-    }
-    if (viter != m_vmap.end())
-      return viter->second;
-    else {
+    const auto realNameAndVersion = getRealNameFromAlias(algorithmName);
+    if (realNameAndVersion != boost::none) {
+      return realNameAndVersion->second;
+    } else {
```

Comment:
```suggestion
    if (const auto realNameAndVersion = getRealNameFromAlias(algorithmName)) {
      return realNameAndVersion->second;
    } else {
```

Reply:
is this suggestion reliant on me changing it to use `std::optional`?

Reply:
I think this should work for `boost::optional` too, but I might be wrong!

Reply:
I'll do both changes anyway :)